roles/openshift_health_checker/openshift_checks/package_availability.py

```diff
@@ -0,0 +1,69 @@
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException
+from openshift_checks.mixins import NotContainerized
+
+
+class PackageAvailability(NotContainerized, OpenShiftCheck):
+    """Check that required RPM packages are available."""
+
+    name = "package_availability"
+    tags = ["preflight"]
+
+    def run(self, tmp, task_vars):
+        try:
+            rpm_prefix = task_vars["openshift"]["common"]["service_type"]
+        except (KeyError, TypeError):
+            raise OpenShiftCheckException("'openshift.common.service_type' is undefined")
```

Comment:
Up to now the interface follows from `OpenShiftCheck`: you need to provide a name and a `run` method with the same signature as an action plugin and a module call (tmp and task_vars are input, the return value is a dict). The semantics are also the same as action plugin/module.

Yes, taking `task_vars` in smells like a violation of the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter), but we're just mimicking how action plugins / modules / Ansible works, I haven't found a better way.

If we would take only the values each check requires, then they would not have a common API, and code would need to know how to instantiate each check with a different set of arguments, and that would put us in a worse situation IMHO.

If we are moving away from `openshift_facts` (are we?) then I expect many things to break, because the whole code base depends on the ad-hoc structure of variables.